File: app/src/main/java/omniex/nl/omniex/ui/app/order/shipping/method/OrderShippingMethodPresenter.java
package omniex.nl.omniex.ui.app.order.shipping.method;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import nl.omniex.omniexshopping.data.model.response.ShippingMethodResponse;
import nl.omniex.omniexshopping.data.model.shipping.ShippingMethod;
import nl.omniex.omniexshopping.data.model.shipping.ShippingMethodSetter;
import nl.omniex.omniexshopping.data.repository.ShippingRepository;
import nl.omniex.omniexshopping.ui.base.BasePresenter;

public class OrderShippingMethodPresenter extends BasePresenter<OrderShippingMethodView> {

    private ShippingRepository mShippingRepository;

    @Inject
    OrderShippingMethodPresenter(ShippingRepository shippingRepository) {
        mShippingRepository = shippingRepository;
    }

    void getShippingMethods() {
        addDisposable(mShippingRepository
                .getShippingMethods()
                .subscribe(
                        shippingMethodResponse -> {
                            if (shippingMethodResponse.code() == 200) {
                                ifViewAttached(view -> view.onShippingMethodsFetched(createList(shippingMethodResponse.body())));
                            }
                        }, Throwable::printStackTrace));
    }

    private List<ShippingMethod> createList(ShippingMethodResponse shippingMethodResponse) {
        List<ShippingMethod> shippingMethods = new ArrayList<>();
        shippingMethods.add(shippingMethodResponse.getShippingMethodWrapper().getShippingMethod());
        return shippingMethods;
    }

    void setShippingMethod(ShippingMethodSetter shippingMethodSetter) {
        addDisposable(mShippingRepository
                .setShippingMethod(shippingMethodSetter)
                .subscribe(
                        response -> {
                            if (response.code()==200)
                                ifViewAttached(view -> view.onShippingMethodSet());
                        }, Throwable::printStackTrace));
    }
}

File: app/src/main/java/omniex/nl/omniex/ui/app/order/shipping/method/OrderShippingMethodPresenter.kt
package omniex.nl.omniex.ui.app.order.shipping.method

import java.util.ArrayList

import javax.inject.Inject

import nl.omniex.omniexshopping.data.model.response.ShippingMethodResponse
import nl.omniex.omniexshopping.data.model.shipping.ShippingMethod
import nl.omniex.omniexshopping.data.model.shipping.ShippingMethodSetter
import nl.omniex.omniexshopping.data.repository.ShippingRepository
import nl.omniex.omniexshopping.ui.base.BasePresenter

class OrderShippingMethodPresenter @Inject
internal constructor(private val mShippingRepository: ShippingRepository) : BasePresenter<OrderShippingMethodView>() {

    internal fun getShippingMethods() {
        addDisposable(mShippingRepository
                .getShippingMethods()
                .subscribe(
                        { shippingMethodResponse ->
                            if (shippingMethodResponse.code() === 200) {
                                ifViewAttached { view -> view.onShippingMethodsFetched(createList(shippingMethodResponse.body())) }
                            }
                        }, ???({ printStackTrace() })))
    }

    private fun createList(shippingMethodResponse: ShippingMethodResponse): List<ShippingMethod> {
        val shippingMethods = ArrayList<ShippingMethod>()
        shippingMethods.add(shippingMethodResponse.getShippingMethodWrapper().getShippingMethod())
        return shippingMethods
    }

    internal fun setShippingMethod(shippingMethodSetter: ShippingMethodSetter) {
        addDisposable(mShippingRepository
                .setShippingMethod(shippingMethodSetter)
                .subscribe(
                        { response ->
                            if (response.code() === 200)
                                ifViewAttached { view -> view.onShippingMethodSet() }
                        }, ???({ printStackTrace() })))
    }
}
